### DISCUSSION TITLE
Difference between `cx.render(rsx!` and `render!`

### DISCUSSION THREAD

**Discussion by tkr-sh** [DioxusLabs/dioxus#1831] [General]
- What is the difference between  `cx.render(rsx!` and `render!` ?
- When should we use which ?

**Comment by ealmloff** [comment]
> * What is the difference between  `cx.render(rsx!` and `render!` ?
> * When should we use which ?

They are almost identical, `render!` can be slightly lower overhead because it avoids boxing a closure but it assumes that you have a scope variable named cx. I use render! whenever I have a scope variable named cx
